src/NodeServices/Interfaces.cs

```diff
@@ -141,6 +141,21 @@ internal interface IExecutionSession
         string SearchFile(string filename);
     }
 
+    /// <summary>
+    /// An options structure used when calling IExtensionApplication.Startup
+    /// </summary>
+    public struct ExtensionStartupParams
```

Comment:
any problem with making this internal or it needs to be public to be used as a public parameter?

Reply:
needs to be public to be used as a public parameter?